src/m365/viva/commands/engage/engage-community-add.spec.ts

```diff
@@ -0,0 +1,329 @@
+    assert(loggerLogSpy.calledWith(operationLocation));
+  });
+
+  it('creates a community with adminEntraIds and waits for provisioning to complete', async () => {
+    let waitsForCompletion = false;
+    let i = 0;
+
+    sinon.stub(request, 'post').callsFake(async (opts: any) => {
+      if (opts.url === `https://graph.microsoft.com/beta/employeeExperience/communities`) {
+        return {
+          headers: {
+            location: operationLocation
+          }
+        };
+      }
+      throw 'Invalid request';
+    });
+
+    sinon.stub(request, 'get').callsFake(async (opts: any) => {
+      if (opts.url === operationLocation) {
+        if (i++ < 2) {
+          return {
+            status: 'inprogress'
+          };
+        }
+
+        waitsForCompletion = true;
+        return {
+          status: 'succeeded'
+        };
+      }
+      throw 'Invalid request';
+    });
+
+    sinon.stub(global, 'setTimeout').callsFake((fn) => {
+      fn();
+      return {} as any;
+    });
+
+    await command.action(logger, {
+      options: {
+        displayName: "Software engineers",
+        description: "A community for all software engineer",
+        privacy: "public",
+        adminEntraIds: "50674d84-6bf1-470b-89b5-d55ce0a5a720",
+        wait: true,
+        verbose: true
+      }
+    } as any);
+    assert.strictEqual(waitsForCompletion, true);
+  });
+
+  it('creates a community with adminEntraUserNames and waits for provisioning to complete', async () => {
+    let waitsForCompletion = false;
+    let i = 0;
+
+    sinon.stub(request, 'post').callsFake(async (opts: any) => {
+      if (opts.url === `https://graph.microsoft.com/beta/employeeExperience/communities`) {
+        return {
+          headers: {
+            location: operationLocation
+          }
+        };
+      }
+      throw 'Invalid request';
+    });
+
+    sinon.stub(request, 'get').callsFake(async (opts: any) => {
+      if (opts.url === operationLocation) {
+        if (i++ < 2) {
+          return {
+            status: 'inprogress'
+          };
+        }
+
+        waitsForCompletion = true;
+        return {
+          status: 'succeeded'
+        };
+      }
+
+      if (opts.url.startsWith(`https://graph.microsoft.com/v1.0/users/`) && opts.url.endsWith(`?$select=id`)) {
+        return { id: '50674d84-6bf1-470b-89b5-d55ce0a5a720' };
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinon.stub(global, 'setTimeout').callsFake((fn) => {
+      fn();
+      return {} as any;
+    });
+
+    await command.action(logger, {
+      options: {
+        displayName: "Software engineers",
+        description: "A community for all software engineer",
+        privacy: "public",
+        adminEntraUserNames: "john.doe@consoto.onmicrosoft.com",
+        wait: true,
+        debug: true
+      }
+    } as any);
+    assert.strictEqual(waitsForCompletion, true);
+  });
+
+  it('handles error when waiting for provisioning to complete fails', async () => {
```

Comment:
Let's also add an extra test for general Graph API failure requests (while creating the group)